proposals/0000-splitting-out-stable-interfaces-from-th.rst

```diff
@@ -0,0 +1,335 @@
+If we can convince their maintainers to depend on these packages instead, then GHC would no longer (transitively) depend on ``template-haskell``.
+This makes it possible for packages to depend on the ``ghc`` library at the same time as a version of ``template-haskell`` different to the one bundled with that GHC.
+
+
+Proposed Change Specification
+-----------------------------
+No changes to the language or the compiler are required for this proposal.
+
+Proposed Library Change Specification
+-------------------------------------
+
+We propose to publish two new libraries: ``template-haskell-lift`` and ``template-haskell-quasiquote``.
+These will be shipped with GHC. So, they would be boot libraries, but wouldn't include any wired-in identifiers.
+In other words, they would behave as ``bytestring`` or ``containers``, not like ``ghc-internal``.
+
+They will also be published to and buildable from Hackage.
+They can be built with the version of GHC they are bundled with, but should additionally be buildable with the previous and next version of GHC.
+Concretely if ``template-haskell-0.1`` is shipped with GHC-9.14, then there should be minor releases (``template-haskell-0.1.*``) that can be built with GHC-9.12 and GHC-9.16.
+We wish to have as broad a support range as feasible, eg, the current version of the libraries are compatible with GHC-8.10 up to GHC-9.12 (the present release).
+We acknowledge that this might not always be possible, and that these interfaces might need to change in the future in ways that cannot be shimmed over.
+
+These packages only depend on ``ghc-internal`` and ``base``. Crucially they do not depend on ``template-haskell``.
+
+Their initial interfaces will be as follows:
+
+``template-haskell-lift``::
+
+   module TemplateHaskell.Lift
+    ( Q
+    , Code
+    , Quote
+    , Exp
+    , Lift(..)
+    , defaultLiftTyped -- a utility for writing `liftTyped` methods when an instance currently defines lift only
+    , liftAddrCompat -- a utility for creating an `Addr#` value, eg, for defining `Lift ByteString`
+    , liftIntCompat -- a utility for lifting an `Int` without causing issues when used with `OverloadedSyntax`
+    )
+
+``template-haskell-quasiquote``::
+
+   module TemplateHaskell.Quasiquoter
+    ( Q
+    , Exp
+    , Pat
+    , Type
+    , Dec
+    , QuasiQuoter (QuasiQuoter, quoteExp, quotePat, quoteType, quoteDec)
+    )
+
+Note that these modules are in the ``TemplateHaskell.`` namespace rather than the ``Language.Haskell.TH.`` namespace.
+The idea to use this less verbose namespace for the new stable interfaces is thanks to Adam Gundry.
+
+Effect and Interactions
+-----------------------
+This works towards removing the special case for ``template-haskell`` in `(GR1) <https://github.com/ghc-proposals/ghc-proposals/blob/master/principles.rst#33stability-gr1>`_ from `Principles for GHC <../principles.rst>`_, but on its own it doesn't achieve it.
```

Comment:
(Non-actionable in this proposal.)

I just had a look at (GR1) and don't understand why `template-haskell` warrants a special case at all. Perhaps it used to when it was wired-in, but that's not the case any more since it nowadays re-exports from `ghc-internal`. The versioning of the latter accounts for the alluded breaking changes to GHC's AST.

Reply:
Yes we should reexamine GR1 and see if we can drop it